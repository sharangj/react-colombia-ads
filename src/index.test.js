import React from 'React'
import renderer from 'react-test-renderer'
import { ColombiaAd } from './index.js'

describe('ColombiaAd', () => {
  it('is truthy', () => {
    expect(ColombiaAd).toBeTruthy()
  })

  it('does render colombia script', () => {
    const tree = renderer
      .create(
        <ColombiaAd
          mobileSlotId='<mobile id>'
          desktopSlotId='<desktop id>'
          section='ArticleShow'
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders only mobile when desktop slot id is not passed', () => {
    const tree = renderer
      .create(<ColombiaAd mobileSlotId='<mobile id>' section='ArticleShow' />)
      .toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree.props.className).toBe('displayMobile')
  })

  it('renders only desktop script for when mobile slot id is not passed', () => {
    const tree = renderer
      .create(<ColombiaAd desktopSlotId='<desktop id>' section='ArticleShow' />)
      .toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree.props.className).toBe('displayDesktop')
  })
})
