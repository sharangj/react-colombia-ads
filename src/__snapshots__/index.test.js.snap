// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColombiaAd does render colombia script 1`] = `
Array [
  <div
    className="displayMobile"
  >
    <div
      className="colombia"
      data-position="1"
      data-section="ArticleShow"
      data-slot="<mobile id>"
      data-ua="M"
      id="div-clmb-ctn-<mobile id>-1"
      style={
        Object {
          "minHeight": "2px",
          "width": "100%",
        }
      }
    />
  </div>,
  <div
    className="displayDesktop"
  >
    <div
      className="colombia"
      data-position="2"
      data-section="ArticleShow"
      data-slot="<desktop id>"
      data-ua="D"
      id="div-clmb-ctn-<desktop id>-1"
      style={
        Object {
          "minHeight": "2px",
          "width": "100%",
        }
      }
    />
  </div>,
]
`;

exports[`ColombiaAd renders only desktop script for when mobile slot id is not passed 1`] = `
<div
  className="displayDesktop"
>
  <div
    className="colombia"
    data-position="2"
    data-section="ArticleShow"
    data-slot="<desktop id>"
    data-ua="D"
    id="div-clmb-ctn-<desktop id>-1"
    style={
      Object {
        "minHeight": "2px",
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`ColombiaAd renders only mobile when desktop slot id is not passed 1`] = `
<div
  className="displayMobile"
>
  <div
    className="colombia"
    data-position="1"
    data-section="ArticleShow"
    data-slot="<mobile id>"
    data-ua="M"
    id="div-clmb-ctn-<mobile id>-1"
    style={
      Object {
        "minHeight": "2px",
        "width": "100%",
      }
    }
  />
</div>
`;
